app/repository/helpers.py

```diff
@@ -10,40 +10,62 @@
 from sqlalchemy.orm import Session
 
 
-def generate_slug(
-    db: Session,
-    title: str,
-    attempts: int = 100,
-    suffix_length: int = 4,
+def generate_unique_slug(
+    existing_slugs: set[str], title: str, attempts: int = 100, suffix_length: int = 6
 ) -> str:
     """
-    Generates a slug that doesn't already exists.
+    Generates a slug that doesn't already exist.
 
-    :param Session db: The connection to the db
-    :param str title: The title or name of the object you wish to slugify
+    :param set[str] existing_slugs: A set of already existing slugs to validate against.
+    :param str title: The title or name of the object you wish to slugify.
     :param int attempts: The number of attempt to generate a unique slug before
-    failing, defaults to 100
-    :param int suffix_length: The suffix to produce uniqueness, defaults to 4
-    :raises RuntimeError: If we cannot produce a unique slug
-    :return str: the slug generated
+    failing, defaults to 100.
+    :param int suffix_length: The suffix to produce uniqueness, defaults to 6.
+    :raises RuntimeError: If we cannot produce a unique slug.
+    :return str: The slug generated.
     """
-    lookup = set([cast(str, n) for n in db.query(Slug.name).all()])
     base = slugify(str(title))
 
     # TODO: try to extend suffix length if attempts are exhausted
     suffix = str(uuid4())[:suffix_length]
     count = 0
-    while (slug := f"{base}_{suffix}") in lookup:
+    while (slug := f"{base}_{suffix}") in existing_slugs:
```

Comment:
don't we want to check  a combination of `lookup` (slugs from the db) and `existing` (slugs part of the transaction)

Reply:
Yes, that's why they get combined in the function that calls this one so this function still only gets a single set with all slugs. See generate_slug()